fearless/Modules/Staking/Analytics/AnalyticsValidators/ViewModel/AnalyticsValidatorsViewModelFactory.swift

```diff
@@ -0,0 +1,287 @@
+import SoraFoundation
+import FearlessUtils
+import BigInt
+import IrohaCrypto
+
+final class AnalyticsValidatorsViewModelFactory: AnalyticsValidatorsViewModelFactoryProtocol {
+    private lazy var iconGenerator = PolkadotIconGenerator()
+
+    private let balanceViewModelFactory: BalanceViewModelFactoryProtocol
+    private let chain: Chain
+    private let percentFormatter = NumberFormatter.percent
+
+    init(
+        balanceViewModelFactory: BalanceViewModelFactoryProtocol,
+        chain: Chain
+    ) {
+        self.balanceViewModelFactory = balanceViewModelFactory
+        self.chain = chain
+    }
+
+    func createViewModel(
+        eraValidatorInfos: [SubqueryEraValidatorInfo],
+        stashAddress: AccountAddress,
+        rewards: [SubqueryRewardItemData],
+        nomination: Nomination,
+        identitiesByAddress: [AccountAddress: AccountIdentity]?,
+        page: AnalyticsValidatorsPage
+    ) -> LocalizableResource<AnalyticsValidatorsViewModel> {
+        LocalizableResource { locale in
+            self.percentFormatter.locale = locale
+
+            let totalEras = self.totalErasCount(eraValidatorInfos: eraValidatorInfos)
```

Comment:
Maybe create a method which returns set of distinct eras than the total count. In this case we could reuse them for examples on line 34